### DISCUSSION TITLE
关于从零开始的智能网关部署

### DISCUSSION THREAD

**Discussion by someone132s** [XiaoMi/ha_xiaomi_home#180] [Q&A]
想问一下，我现在没有任何智能设备的情况下，想使用这个方案起码需要一个ha，一个小米中枢网关，然后使用文章中的插件纳管这个中枢网关？那现在还有几个问题。首先是小米的多模网关，它是需要被中枢网关纳管还是可以直接由ha纳管？小米的智能设备可以通过第三方网关设备连接ha或者中枢网关么？比如我手搓一个zigbee网关出来？最后就是小米中枢网关是支持热备的，那ha那边需要进行额外配置么？如果我的ha也配置热备，一旦ha切换之后，中枢网关能立刻切过去么？

**Comment by JimmyKmi** [comment]
小白用户不建议折腾HA，可以先去B站了解米家中枢主从架构。

**Reply by someone132s** [reply]
明白了，智能家居这一块我确实完全没有接触过。我先去看看再说吧。

**Reply by someone132s** [reply]
@JimmyKmi 不对啊主从和热备是两码事

**Reply by Ghost-Sam1222** [reply]
学习顺序如何呢？我是小白入手米家的时候嫌他的自动化太智障，就了解到了home assistant照着ai写自动化确实效率低，一看好多别的设备都有集成可以接入，没有集成的也有nodered这个我就比较喜欢了。况且我家是iOS生态也有appletv，homekit偏好

**Comment by TioaChan** [comment]
1. 只需要HomeAssistant即可使用本集成
2. 本集成和米家App在地位上是一样的，可以和App一样控制设备，但不能添加设备，只要能在米家App上看到设备，在本集成上也能看到

https://github.com/XiaoMi/ha_xiaomi_home/blob/main/doc/README_zh.md#%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98

**Reply by someone132s** [reply]
感谢您的回复。

我大概明白了这个插件的作用，其实就是将小米智能中枢的管理接口暴露给ha用这样么？那这样也就不需要通过小米的服务器了么？还是说我ha管理小米的设备还是需要ha先跟云服务器通信，然后云服务器再下发指令呢？
由于离线部署对我来说是关键，不能离线使用就没有任何意义了。（当然部署阶段使用在线服务是应该的）

感谢解答。

**Reply by TioaChan** [reply]
米家智能设备分为三种，Wi-Fi设备，蓝牙/蓝牙mesh设备，zigbee设备。按本集成readme的说法，本集成可以通过控制中枢网关来实现本地化控制设备，或直接通过局域网直接控制大部分Wi-Fi设备。如果没有中枢网关，需要通过云端来控制蓝牙/蓝牙mesh设备，前提是需要有其他类型的网关在网。

现阶段有其他集成可以不使用中枢网管，仅多模网关也能本地控制蓝牙/蓝牙mesh/zigbee设备。

Home Assistant的目的就是跨平台统一管理智能设备。

**Reply by L1yp** [reply]
TC哥，你好，我今天测试发现中枢网关断网后，HA中中枢网关的虚拟事件就不可用了，虚拟事件是依赖云端的吗？

**Comment by Hugh696** [comment]
首先不要管这个集成，在米家添加全部的智能设备，多模网关作为从网关，是由中枢网关管理的。
自动化和联动直接在米家app写，不够的就在中枢的米家极客版写，这两步的自动化都是保存在中枢网关中，可能有的会直接保存在多模网关中；
这个集成是镜像整个米家app的设备，前面两步都无法做到的自动化才考虑在Home Assistant里写，这样万一网络出故障，损失才会最低。
你说的手搓zigbee网关，是可以接入小米的部分zigbee子设备，接入到Home Assistant 中，但是这本身和米家app毫无关联，也就跟此集成毫无关联。

**Reply by someone132s** [reply]
感谢你的回复。我之前不入坑智能家居的原因就是因为米家不开放个人开发者权限，也没有一个很理想的方案可以跨平台的整合各家设备。所以对我来说，要部署智能家居就肯定是部署到ha上面的，我不会打算去使用米家的智能功能。

我希望小马的智能中枢只要提供接入即可。所有的管理都使用ha完成。我甚至还希望米家智能中枢能使用虚拟化部署，或者容器部署的方案。这样管理起来也更加方便一点。

我对家里的网络还是比较有信心，相信是不会垮掉的。

现在最为担心的还是这个插件和米家智能中枢的可靠性问题，毕竟这个要承载全家的用电等等，不做个双活不太敢用。万一哪天挂了岂不是灯都开不了。

最后zigbee部分我大概明白了，能直接接入ha当然更好，不需要涉及米家也是我希望看到的。

**Reply by JimmyKmi** [reply]
> 感谢你的回复。我之前不入坑智能家居的原因就是因为米家不开放个人开发者权限，也没有一个很理想的方案可以跨平台的整合各家设备。所以对我来说，要部署智能家居就肯定是部署到ha上面的，我不会打算去使用米家的智能功能。
> 
> 我希望小马的智能中枢只要提供接入即可。所有的管理都使用ha完成。我甚至还希望米家智能中枢能使用虚拟化部署，或者容器部署的方案。这样管理起来也更加方便一点。
> 
> 我对家里的网络还是比较有信心，相信是不会垮掉的。
> 
> 现在最为担心的还是这个插件和米家智能中枢的可靠性问题，毕竟这个要承载全家的用电等等，不做个双活不太敢用。万一哪天挂了岂不是灯都开不了。
> 
> 最后zigbee部分我大概明白了，能直接接入ha当然更好，不需要涉及米家也是我希望看到的。

我觉得你可能更需要一些Matter设备而不是米家设备，在对设备配网已经设备插件拉取的时候仍需要进行联网，而对设备进行设置时在米家APP对设备进行配置是必须的（大多数设备无法通过远程修改配置，或有些配置项并不能通过spec修改）

中枢在遵循官方建议下使用可靠性是极佳的，但无法阻止用户出现（但不限于）下列行为：
- 不看警告，将中枢与路由器堆叠导致蓝牙信号芯片烧毁
- 家里DNS强制劫持而且还有问题
- 用poe给中枢供电但转换模块质量有问题导致供电不足
- 把中枢丢在全金属弱电箱说信号不好
- 使用1ms高频循环执行设备指令

此外网络问题涉及因素过多，《我家网络绝对没问题》的因素包括但不限于：
- 开启多频合一，但5G信号不稳定导致设备频繁从2.4G和5G频段疯狂跳转
- 路由信道不固定导致信道切换时设备掉线
- 报文限制过小或过大
- 开启了防蹭网并间歇性将设备踢下线
- 开启mesh组网时设备在不同路由器间跳换
- 蓝牙网关密度不合理或导致覆盖区过度重叠或信号不佳
再者我还见过：
- AP供电不足但AC没有任何提示，出现问题全然不知的
- 交换机包转发不过来的
- 线路出现虚接导致链路频繁重协商的
上述问题大多在手机上网时不会有任何体感，但很多会导致IOT设备经常性掉线。但问题不仅限于上述问题。

**Reply by someone132s** [reply]
明白您的意见了，我对matter设备确实没有什么了解，仅仅是知道它可以接入ha。而且看下来matter目前市场占有率似乎不是很高。我去搜搜了解一下看看。

我觉得在设置部分使用米家是没有问题的，我希望能实现的是，实际应用时所有的设备都能由ha在本地完成管理。在添加设备等等的时候，使用米家来完成我是完全可以接受的。我不能接受比如说灯是ha管理，其他传感器却只能在米家上看，它们之间不能联动的这样的情况；或者是ha需要通过云给米家下发控制指令的情况。

至于中枢的可靠性问题，能听到它很稳定就太好了。我留意到中枢有热备功能，不知道这个功能能不能和ha搭配使用。而且如果给ha也部署热备功能，中枢能不能对接的上呢？希望能够解答，感谢了！

最后提到的中枢的部署方式，考虑到我有poe交换机，我确实希望可以用poe的方法部署中枢，并且也希望可以部署在机柜内。所以早在提问之前我就大概看过中枢的PCB，应该是可以把蓝牙天线飞出来使用的，如果不行我就放机柜外面吧。

至于网络部分，我使用的是ftnt的全套soho解决方案，目前用下来不到一年的时间里，没有遇到过出问题的情况，也没有碰到不稳定的时候。当然说它稳定或许还有点早，毕竟我的lic是最低级的...，不过我也不可能立刻就讲ha投入使用，再给一年时间观察一下网络是否靠谱也刚好能与对智能家居是否适合我的测试在时间上对的上。（目前我是打算将ha和智能中枢连接到现在使用网关上再单独划个VLAN出来，以后真的用起来再迁移到专门的设备上去吧。）

**Reply by Hugh696** [reply]
> 感谢你的回复。我之前不入坑智能家居的原因就是因为米家不开放个人开发者权限，也没有一个很理想的方案可以跨平台的整合各家设备。所以对我来说，要部署智能家居就肯定是部署到ha上面的，我不会打算去使用米家的智能功能。
> 
> 我希望小马的智能中枢只要提供接入即可。所有的管理都使用ha完成。我甚至还希望米家智能中枢能使用虚拟化部署，或者容器部署的方案。这样管理起来也更加方便一点。
> 
> 我对家里的网络还是比较有信心，相信是不会垮掉的。
> 
> 现在最为担心的还是这个插件和米家智能中枢的可靠性问题，毕竟这个要承载全家的用电等等，不做个双活不太敢用。万一哪天挂了岂不是灯都开不了。
> 
> 最后zigbee部分我大概明白了，能直接接入ha当然更好，不需要涉及米家也是我希望看到的。

米家所有东西都能在Home Assistant中查看管理，而且是本地的，不依赖米家云服务器，任何自动化都能写，只是我不建议全都在Home Assistant里写。
一个很简单的问题，所有自动化都在Home Assistant 中设置固然很统一，但是路径会非常长，哪怕中间不会出现故障，延迟也会比较大。用一个基础的“无线按键控制智能灯（都是zigbee）”的自动化举例，如果你在米家app中设置，最理想的情况下，信息会在“按键-多模网关-灯”这条路径传输，如果你在Home Assistant 中设置，就变成了“按键-多模网关-路由器-中枢网关-路由器-Home Assistant设备-路由器-中枢网关-路由器-多模网关-灯”，想想都可怕。
中枢的热备功能指两台中枢相互备份米家和米家极客版中设置的自动化，一台离线了，另一台顶上。这个功能和米家集成和Home Assistant 也毫无关系，所以更应该把基础的自动化放在米家来写。

**Comment by Ghost-Sam1222** [comment]
我觉得依赖的是中枢本地存储的虚拟事件吧，如果里面采用的动作有需要走云端的操作（例如：操作美的设备）那就走云端，如果没有应该就是全本地
________________________________
